README.md

```diff
@@ -1147,6 +1148,64 @@ configs.gatewayConfigs.Fetch = {
 
 Take a look [here](https://github.com/tulios/mappersmith/blob/master/src/mappersmith.js) for more options.
 
+### <a name="typescript"></a> TypeScript
+
+__Mappersmith__ also supports TypeScript (>=3.5). In the following sections there are some common examples for using TypeScript with Mappersmith where it is not too obvious how typings are properly applied.
+
+#### Create a middleware with TypeScript
+
+To create a middleware using TypeScript you just have to add the `Middleware` interface to your middleware object:
+
+```typescript
+import { Middleware } from 'mappersmith'
+
+const MyMiddleware: Middleware = () => ({
+  prepareRequest(next) {
+    return next().then(response => response.enhance({
```

Comment:
`return next().then(request => request.enhance({`

small typo, it should be `request`.

Reply:
Ohh yes, changed it!